### PR TITLE
Rename merge to Mode. Implement SEQ, MD, Merge save modes

### PR DESCRIPTION
**Pull request by jmamma**

### CODE REVIEW

File: avr/cores/megacommand/MCL/GridSavePage.cpp [hunk 1]
```diff
@@ -38,12 +39,20 @@ void GridSavePage::display() {
   GUI.setLine(GUI.LINE2);
   GUI.put_string_at_fill(0, "S");
 
-  if ((MidiClock.state != 2) && (encoders[0]->cur == 1)) {
-    GUI.put_string_at(3, "MERGE");
+  const char *merge = "SEQ";
+  if (MidiClock.state == 2) {
+    merge = "---";
```
Comment (yatli) [inline avr/cores/megacommand/MCL/GridSavePage.cpp:44]: Should be "seq" now?

**Comment by yatli** [comment]
I'll update the data flow visualization later today.

**Comment by yatli** [comment]
The graphics is updated.

**Comment by yatli** [comment]
Updated the doc.

**Comment by jmamma** [comment]
Thanks @yatli 

I need to test this before merge.

**Comment by yatli** [comment]
How are the parameter locks dumped/merged? I notice it's different for a pattern on MD vs. what is saved to MC.

**Comment by jmamma** [comment]
There are 3 representations for parameter locks.

MDPattern
MDTrack
MDSeqTrackData


The main code to extract parameter locks from MDPattern is in MDTrack
```
  int n = 0;
  arraysize = 0;
  for (int i = 0; i < 24; i++) {
    if (IS_BIT_SET32(MD.pattern.lockPatterns[tracknumber], i)) {
      int8_t idx = MD.pattern.paramLocks[tracknumber][i];
      if (idx >= 0) {
        for (int s = 0; s < 64; s++) {

          if ((MD.pattern.locks[idx][s] <= 127) &&
              (MD.pattern.locks[idx][s] >= 0)) {
            if (IS_BIT_SET64(trigPattern, s)) {

              locks[n].step = s;
              locks[n].param_number = i;
              locks[n].value = MD.pattern.locks[idx][s];
              n++;
            }   
          }   
        }   
      }   
    }   
  }

  //  itoa(n,&str[2],10);

  arraysize = n;
```

The code to merge MDTrack locks in to MDSeqTrackData 

```
void MDSeqTrack::merge_from_md(MDTrack *md_track) {
  DEBUG_PRINT_FN();

  if (md_track->trigPattern == 0 && ((pattern_mask | lock_mask) != 0)) {
    // If the MD sequencer data is empty, abort merge.
    // This will prevent unnecessary length change of internal seq pattern
    return;
  }
  if ((pattern_mask | lock_mask) == 0) {
    set_length(md_track->length);
  }

  for (int n = 0; n < md_track->arraysize; n++) {
    set_track_locks(md_track->locks[n].step, md_track->locks[n].param_number,
                    md_track->locks[n].value);
    SET_BIT64(lock_mask, md_track->locks[n].step);
  }

**Comment by jmamma** [comment]
Patches above should fix merging.
Brief tests show it's okay.
Fixed issue with row names.

Needs more testing.

**Comment by jmamma** [comment]
merged for now.

**Pull request merged**